processor/resourceprocessor/resource_processor.go

```diff
@@ -0,0 +1,101 @@
+// Copyright 2019 OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package resourceprocessor
+
+import (
+	"context"
+
+	resourcepb "github.com/census-instrumentation/opencensus-proto/gen-go/resource/v1"
+
+	"github.com/open-telemetry/opentelemetry-collector/consumer"
+	"github.com/open-telemetry/opentelemetry-collector/consumer/consumerdata"
+	"github.com/open-telemetry/opentelemetry-collector/processor"
+)
+
+type resourceTraceProcessor struct {
+	resource *resourcepb.Resource
+	next     consumer.TraceConsumer
+}
+
+func newResourceTraceProcessor(next consumer.TraceConsumer, cfg *Config) *resourceTraceProcessor {
+	resource := createResource(cfg)
+	return &resourceTraceProcessor{
+		next:     next,
+		resource: resource,
+	}
+}
+
+// ConsumeTraceData implements the TraceProcessor interface
+func (rtp *resourceTraceProcessor) ConsumeTraceData(ctx context.Context, td consumerdata.TraceData) error {
+	return rtp.next.ConsumeTraceData(ctx, consumerdata.TraceData{
+		Node:         td.Node,
+		Resource:     rtp.resource,
```

Comment:
shouldn't this be adding to `td.Resource`? In case of conflict I think it is fair for it to overwrite any key but other stages in the pipe can be adding resources and we don't want to drop them.

Reply:
Agree. We should be merging.

Reply:
Also, if we don't merge, should we be copying this instead? If we just assign, another processor down the pipeline could end up modifying the static resource?

Reply:
Sounds good. The main issue is what happens when the resource types aren't the same - in that case, we can either:

1. overwrite with the new resource type and merge the labels
2. replace with the new resource rather than merge
3. keep the original resource and drop the new one

I've gone with option 1, let me know if you have any preferences or other options (If we had an ordering on resource types, we could make a decision based on that but I don't think that's something we want to get into at this point).

Reply:
I think existing processors that do similar work follow 1. They replace any pre-existing conflicting labels.